Mathlib/Analysis/Analytic/Prod.lean

```diff
@@ -0,0 +1,97 @@
+/-
+Copyright (c) 2023 Geoffrey Irving. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Geoffrey Irving
+-/
+import Mathlib.Analysis.Analytic.Composition
+import Mathlib.Analysis.Analytic.Linear
+
+/-!
+# Analytic functions on and to product spaces
+-/
+
+noncomputable section
+
+open Topology Classical BigOperators NNReal Filter ENNReal
+
+open Function (uncurry)
+
+open Set Filter Asymptotics
+
+variable {𝕜 E F G H : Type*}
+
+variable [NontriviallyNormedField 𝕜] [NormedAddCommGroup E] [NormedSpace 𝕜 E] [NormedAddCommGroup F]
+  [NormedSpace 𝕜 F] [NormedAddCommGroup G] [NormedSpace 𝕜 G] [NormedAddCommGroup H]
+  [NormedSpace 𝕜 H]
+
+/-- Products of analytic functions are analytic -/
+theorem AnalyticAt.prod {f : E → F} {g : E → G} {x : E} (fa : AnalyticAt 𝕜 f x)
+    (ga : AnalyticAt 𝕜 g x) : AnalyticAt 𝕜 (fun x ↦ (f x, g x)) x := by
+  rcases fa with ⟨p, pr, fp⟩
+  rcases ga with ⟨q, qr, gq⟩
+  set pq : FormalMultilinearSeries 𝕜 E (F × G) := fun n ↦ (p n).prod (q n)
+  have pqr : min pr qr ≤ pq.radius := by
+    apply ENNReal.le_of_forall_nnreal_lt; intro r rr
+    rcases p.norm_mul_pow_le_of_lt_radius
+        (lt_of_lt_of_le rr (_root_.trans (min_le_left pr qr) fp.r_le)) with
+      ⟨pc, _, ph⟩
+    rcases q.norm_mul_pow_le_of_lt_radius
+        (lt_of_lt_of_le rr (_root_.trans (min_le_right pr qr) gq.r_le)) with
+      ⟨qc, _, qh⟩
+    apply pq.le_radius_of_bound (max pc qc); intro n
+    calc ‖pq n‖ * ↑r^n
+      _ = max ‖p n‖ ‖q n‖ * ↑r^n := by simp only [ContinuousMultilinearMap.op_norm_prod]
+      _ = max (‖p n‖ * ↑r^n) (‖q n‖ * ↑r^n) := max_mul_of_nonneg _ _ (pow_nonneg (coe_nonneg r) n)
+      _ ≤ max pc qc := max_le_max (ph n) (qh n)
+  use pq, min pr qr
+  exact
+    { r_le := pqr
+      r_pos := lt_min fp.r_pos gq.r_pos
+      hasSum := by
+        intro y yr; apply HasSum.prod_mk
+        exact fp.hasSum (EMetric.ball_subset_ball (min_le_left _ _) yr)
+        exact gq.hasSum (EMetric.ball_subset_ball (min_le_right _ _) yr) }
+
+/-- Products of analytic functions are analytic -/
+theorem AnalyticOn.prod {f : E → F} {g : E → G} {s : Set E} (fa : AnalyticOn 𝕜 f s)
+    (ga : AnalyticOn 𝕜 g s) : AnalyticOn 𝕜 (fun z ↦ (f z, g z)) s :=
+  fun _ m ↦ (fa _ m).prod (ga _ m)
+
+/-- `AnalyticAt.comp` for a curried function -/
+theorem AnalyticAt.curry_comp {h : F → G → H} {f : E → F} {g : E → G} {x : E}
```

Comment:
The canonical name per our naming conventions would be
```suggestion
theorem AnalyticAt.comp₂ {h : F → G → H} {f : E → F} {g : E → G} {x : E}
```
Same thing below.

Reply:
Hmm that does seem like it's `defeq` to the version with explicit curry, but I should check to see if things infer correctly when doing `apply` in tactics.  +1 to it being way better if things do work.

(Thank you for this and the other comments!  Will check this and resolve the other changes later, but might be a couple day delay.)

Reply:
I checked, and this works great.  Fixed.